apps/life/models.py

```diff
@@ -183,6 +209,28 @@ def last_known_push(self):
             # no push in repo, return 0
             return 0
 
+    def relative_path(self):
+        '''If this is a unified local clone, find out which repo
+        we're pulling for that.
+        Relative path to the clone, thus the name.
+        '''
+        relative_path = self.name
+        if self.fork_of:
+            relative_path = self.fork_of.name
+        if self.forest and self.forest.fork_of:
+            relative_path = self.forest.fork_of.name + '/' + self.locale.code
+        return relative_path
+
+    def local_path(self):
+        '''If this is a unified local clone, find out which repo
+        we're pulling for that.
+        Used to compute where to search for the local clone, thus
+        the name.
+        '''
+        return os.path.join(
+            settings.REPOSITORY_BASE,
+            *self.relative_path().split('/'))
+
```

Comment:
The `Repository` and `Forest` class could probably inherit from the same parent class.